test/e2e/istio_controller_test_helpers.go

```diff
@@ -341,117 +332,65 @@ func HPAExists(ctx context.Context, t *testing.T, kubeClient client.Client, name
 	return ResourceExists(ctx, t, kubeClient, &hpa, namespace, name)
 }
 
-// TODO this should be a `map[metav1.GroupVersionKind][]types.NamespacedName`
-type ClusterState struct {
-	Istios      []types.NamespacedName
-	Deployments []types.NamespacedName
-	Pods        []types.NamespacedName
-	Services    []types.NamespacedName
-	Hpas        []types.NamespacedName
-}
+type groupVersionResourceString string
+type ClusterResourceList map[groupVersionResourceString][]types.NamespacedName
 
-func clusterIsClean(before ClusterState, after ClusterState) bool {
+func clusterIsClean(before ClusterResourceList, after ClusterResourceList) bool {
 	return reflect.DeepEqual(before, after)
 }
 
 // TODO add more resource types
-func listAllResources(client client.Client) (*ClusterState, error) {
-	istios, err := listIstios(client)
-	if err != nil {
-		return nil, err
-	}
-	deployments, err := listDeployments(client)
-	if err != nil {
-		return nil, err
-	}
-	pods, err := listPods(client)
-	if err != nil {
-		return nil, err
-	}
-	services, err := listServices(client)
-	if err != nil {
-		return nil, err
-	}
-	hpas, err := listHorizontalPodAutoscalers(client)
-	if err != nil {
-		return nil, err
-	}
-	return &ClusterState{istios, deployments, pods, services, hpas}, nil
-}
-
-// TODO rewrite these using the dynamic client
-func listIstios(client client.Client) ([]types.NamespacedName, error) {
-	var istioList istiov1beta1.IstioList
-	err := client.List(context.TODO(), &istioList)
-	if err != nil {
-		return nil, err
-	}
-
-	result := make([]types.NamespacedName, len(istioList.Items))
-	for i, item := range istioList.Items {
-		result[i] = types.NamespacedName{Namespace: item.Namespace, Name: item.Name}
-	}
-	sortNamespacedNames(result)
-	return result, nil
-}
-
-func listDeployments(client client.Client) ([]types.NamespacedName, error) {
-	var deploymentList appsv1.DeploymentList
-	err := client.List(context.TODO(), &deploymentList)
-	if err != nil {
-		return nil, err
+func listAllResources(d dynamic.Interface) (ClusterResourceList, error) {
+	gvrs := []schema.GroupVersionResource{
+		gvr.Service,
+		gvr.Pod,
+		gvr.Deployment,
+		gvr.HorizontalPodAutoscaler,
+		gvr.ClusterRole,
+		gvr.ClusterRoleBinding,
+		gvr.ValidatingWebhookConfiguration,
+		gvr.MutatingWebhookConfiguration,
+		gvr.DestinationRule,
+		gvr.VirtualService,
+		gvr.PeerAuthentication,
+		gvr.Gateway,
+		gvr.EnvoyFilter,
+		gvr.AttributeManifest,
+		gvr.IstioConfigHandler,
+		gvr.IstioConfigInstance,
+		gvr.IstioConfigRule,
```

Comment:
I think these are probably Mixer related resources, I don't think we need these anymore.

Reply:
Yeah, that's what I was thinking too. Shouldn't we just rip the mixer related code out?

Reply:
Actually, I wouldn't spend a minute on that. Mixer won't be in v2, that should be enough. I'd just delete these from here.

Reply:
Yeah, that makes sense.